examples/psbt.rs

```diff
@@ -0,0 +1,711 @@
+    for i in indecies {
+        let pk = pk_path[i].0;
+        let path = pk_path[i].1;
+
+        let pk = PublicKey::from_str(pk).unwrap();
+        let path = path.into_derivation_path().unwrap();
+
+        tree.insert(pk.inner, (fingerprint, path));
+    }
+    tree
+}
+
+/// Does the second update according to the BIP, returns the newly updated PSBT. Verifies against BIP174 test vector.
+fn update_psbt_with_sighash_all(mut psbt: Psbt) -> Psbt {
+    let expected_psbt_hex = include_str!("data/update_2_psbt_hex");
+
+    let ty = PsbtSighashType::from_str("SIGHASH_ALL").unwrap();
+
+    let mut input_0 = psbt.inputs[0].clone();
+    input_0.sighash_type = Some(ty);
+    let mut input_1 = psbt.inputs[1].clone();
+    input_1.sighash_type = Some(ty);
+
+    psbt.inputs = vec![input_0, input_1];
+
+    let got = serialize_hex(&psbt);
+    assert_eq!(got, expected_psbt_hex);
+
+    psbt
+}
+
+/// Verifies the keys in the test vector are valid for the extended private key and derivation path.
+fn parse_and_verify_keys(ext_priv: &ExtendedPrivKey, sk_path: &[(&str, &str)]) -> Vec<PrivateKey> {
+    let secp = SECP256K1;
+
+    let mut keys = vec![];
+    for (secret_key, derivation_path) in sk_path.iter() {
+        let wifi_priv = PrivateKey::from_wif(secret_key).expect("failed to parse key");
+
+        let path = derivation_path.into_derivation_path().expect("failed to convert derivation path");
+        let derived_priv = ext_priv.derive_priv(&secp, &path).expect("failed to derive ext priv key").to_priv();
+        assert_eq!(wifi_priv, derived_priv);
+        keys.push(wifi_priv)
+    }
+    keys
+}
+
+/// Does the first signing according to the BIP, returns the signed PSBT. Verifies against BIP174 test vector.
+fn signer_one_sign(psbt: Psbt, keys: &[PrivateKey]) -> Psbt {
+    let expected_psbt_hex = include_str!("data/sign_1_psbt_hex");
+
+    let psbt = sign(psbt, keys);
+
+    let got = serialize_hex(&psbt);
+    assert_eq!(got, expected_psbt_hex);
+
+    psbt
+}
+
+/// Does the second signing according to the BIP, returns the signed PSBT. Verifies against BIP174 test vector.
+fn signer_two_sign(psbt: Psbt, keys: &[PrivateKey]) -> Psbt {
+    let expected_psbt_hex = include_str!("data/sign_2_psbt_hex");
+
+    let psbt = sign(psbt, keys);
+
+    let got = serialize_hex(&psbt);
+    assert_eq!(got, expected_psbt_hex);
+
+    psbt
+}
+
+/// Does the combine according to the BIP, returns the combined PSBT. Verifies against BIP174 test vector.
+fn combine(mut this: Psbt, that: Psbt) -> Psbt {
+    let expected_psbt_hex = include_str!("data/combine_psbt_hex");
+
+    this.combine(that).expect("failed to combine PSBTs");
+
+    let want_psbt = hex_psbt!(expected_psbt_hex).unwrap();
+    assert_eq!(this, want_psbt);
+
+    // TODO: Something in PSBT is not roundtripping deserialize/serialize, should it?
+    // let got = serialize_hex(&this);
+    // assert_eq!(got, expected_psbt_hex);  // TODO: This call fails.
```

Comment:
This can be removed now, PSBT serialization is not deterministic. See https://github.com/rust-bitcoin/rust-bitcoin/issues/937. I think we should be doing all the `assert_eq!` statements with PSBTs like this one not with hex like all the others do so we do not falsely imply that hex serialization is deterministic.